jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -357,6 +357,15 @@ public HiveConnection(String uri, Properties info) throws SQLException {
           }
 
           if (numRetries >= maxRetries) {
+            try {
+              // Cleaning up the server resources early
+              if (!isClosed) {
+                // close the session
+                close();
+              } else if (transport != null) {
+                transport.close();
+              }
```

Comment:
I think you could also extend the close to be sure to close the transport if that's the only one open